lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -122,18 +122,23 @@ struct LoadOpConversion
 
 #ifdef USE_ROCM
       Value pred = mask ? maskElems[vecStart] : int_val(1, 1);
+      Value zeroVal = bitcast(int_val(valueElemNbits, 0), valueElemTy);
       for (size_t wordIdx = 0; wordIdx < nWords; ++wordIdx) {
         for (size_t wordElem = 0; wordElem < wordNElems; ++wordElem) {
           size_t elemOffset = vecStart + wordIdx * wordNElems + wordElem;
-
-          // get values
-          Value trueVal = load(ptrElems[elemOffset]);
-          Value zeroVal = bitcast(int_val(valueElemNbits, 0), valueElemTy);
-          Value falseVal = other ? otherElems[elemOffset] : zeroVal;
-
-          // select value based on mask
-          Value ret = select(pred, trueVal, falseVal);
-          loadedVals.push_back(ret);
+          Value otherVal = other ? otherElems[elemOffset] : zeroVal;
+          auto *curBlock = rewriter.getBlock();
+          auto *endBlock = curBlock->splitBlock(rewriter.getInsertionPoint());
+          endBlock->addArgument({valueElemTy}, {loc});
```

Comment:
MLIR does not use PHI nodes to enable blocks to communicate values between each other:

https://mlir.llvm.org/docs/Dialects/LLVM/#phi-nodes-and-block-arguments